src/coreclr/src/jit/lsra.cpp

```diff
@@ -2994,291 +2989,584 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
+
+            // Find the next RefPosition of the register.
+            LsraLocation nextIntervalLocation = getNextIntervalRef(bestFitCandidateRegNum, regType);
+            LsraLocation nextPhysRefLocation  = getNextFixedRef(bestFitCandidateRegNum, regType);
+            nextPhysRefLocation               = Min(nextPhysRefLocation, nextIntervalLocation);
+            // If the nextPhysRefLocation is a fixedRef for the rangeEndRefPosition, increment it so that
+            // we don't think it isn't covering the live range.
+            // This doesn't handle the case where earlier RefPositions for this Interval are also
+            // FixedRefs of this regNum, but at least those are only interesting in the case where those
+            // are "local last uses" of the Interval - otherwise the liveRange would interfere with the reg.
+            // TODO: This duplicates code in an earlier loop, and is basically here to duplicate previous
+            // behavior; see if we can avoid this.
+            if (nextPhysRefLocation == rangeEndLocation && rangeEndRefPosition->isFixedRefOfReg(bestFitCandidateRegNum))
+            {
+                INDEBUG(dumpLsraAllocationEvent(LSRA_EVENT_INCREMENT_RANGE_END, currentInterval));
+                nextPhysRefLocation++;
+            }
+
+            if (nextPhysRefLocation == bestFitLocation)
+            {
+                bestFitSet |= bestFitCandidateBit;
+            }
+            else
+            {
+                bool isBetter = false;
+                if (nextPhysRefLocation > lastLocation)
+                {
+                    // This covers the full range; favor it if the other doesn't, or if it's a closer match.
+                    if ((bestFitLocation <= lastLocation) || (nextPhysRefLocation < bestFitLocation))
+                    {
+                        isBetter = true;
+                    }
+                }
+                else
+                {
+                    // This doesn't cover the full range; favor it if the other doesn't either, but this ends later.
+                    if ((bestFitLocation <= lastLocation) && (nextPhysRefLocation > bestFitLocation))
+                    {
+                        isBetter = true;
+                    }
+                }
+                if (isBetter)
+                {
+                    bestFitSet      = bestFitCandidateBit;
+                    bestFitLocation = nextPhysRefLocation;
+                }
+            }
+        }
+        assert(bestFitSet != RBM_NONE);
+        found = selector.applySelection(BEST_FIT, bestFitSet);
+    }
+
+    // Apply the IS_PREV_REG heuristic. Only applies if we have freeCandidates.
+    // Oddly, the previous heuristics only considered this if it covered the range.
+    if ((prevRegRec != nullptr) && ((selector.score & COVERS_FULL) != 0))
+    {
+        found = selector.applySingleRegSelection(IS_PREV_REG, prevRegBit);
```

Comment:
No, I don't think so. If the previous reg was in the candidate set, then `found` will be true after this (because we'll only have that single bit left). Otherwise, if it is not in the set to start with, then `found` will be false.